src/main/scala-2.12/org/apache/james/gatling/imap/scenari/MassiveOperationScenario.scala

```diff
@@ -0,0 +1,78 @@
+package org.apache.james.gatling.imap.scenari
+
+import java.util.Calendar
+
+import com.linagora.gatling.imap.PreDef.{imap, ok}
+import com.linagora.gatling.imap.protocol.command.MessageRange.From
+import com.linagora.gatling.imap.protocol.command.{MessageRanges, Silent, StoreFlags}
+import io.gatling.core.Predef._
+import io.gatling.core.feeder.FeederBuilder
+import io.gatling.core.structure._
+
+import scala.collection.immutable.Seq
+import scala.concurrent.duration._
+
+class MassiveOperationScenario {
+  private val gracePeriod = 5 milliseconds
+  private val numberOfMailInInbox = 20000
+  private val numberOfSubMailboxes = 1000
+  private val mailboxAName = "MailboxA"
+  private val mailboxBName = "MailboxB"
+  private val mailboxCName = "MailboxC"
+  private val mailboxDName = "MailboxD"
+  private val mailboxDNewName = "Dbis"
+
+  private val createMailboxA_B_C = exec(imap("createFolder").createFolder(mailboxAName).check(ok))
+    .exec(imap("createFolder").createFolder(mailboxBName).check(ok))
+    .exec(imap("createFolder").createFolder(mailboxCName).check(ok))
+  private val populateMailboxA = repeat(numberOfMailInInbox)(pause(gracePeriod)
+    .exec(imap("append").append(mailboxAName, Option.empty[Seq[String]], Option.empty[Calendar],
+      """From: expeditor@example.com
+        |To: recipient@example.com
+        |Subject: test subject
+        |
+        |Test content
+        |abcdefghijklmnopqrstuvwxyz
+        |0123456789""".stripMargin).check(ok)))
+  private val setAllMessagesInMailboxASeenFlag = imap("storeAll").store(MessageRanges(From(1L)), StoreFlags.add(Silent.Enable(), "\\Seen")).check(ok)
+  private val copyAllMessagesToMailboxB = imap("copy").copyMessage(MessageRanges(From(1L)), mailboxBName).check(ok)
+  private val moveAllMessagesToMailboxC = imap("move").moveMessage(MessageRanges(From(1L)), mailboxCName).check(ok)
+  private val setAllMessagesInMailboxBDeletedFlagAndExpunge = exec(imap("storeAll").store(MessageRanges(From(1L)), StoreFlags.add(Silent.Enable(), "\\Deleted")).check(ok))
+    .exec(imap("expunge").expunge().check(ok))
+  private val deleteMailboxA_B_C = exec(imap("deleteFolder").deleteFolder(mailboxAName).check(ok))
+    .exec(imap("deleteFolder").deleteFolder(mailboxBName).check(ok))
+    .exec(imap("deleteFolder").deleteFolder(mailboxCName).check(ok))
+  private val createMailboxDWith1000SubMailboxes = exec(imap("createFolder").createFolder(mailboxDName).check(ok))
+    .exec(repeat(numberOfSubMailboxes, "loopId")(pause(gracePeriod).exec(exec(imap("createFolder").createFolder(s"$mailboxDName.$${loopId}").check(ok)))))
+  private val renameMailboxD = exec(imap("renameFolder").renameFolder(mailboxDName, mailboxDNewName).check(ok))
+  private val deleteMailboxD = exec(imap("deleteFolder").deleteFolder(mailboxDNewName).check(ok))
+    .exec(repeat(numberOfSubMailboxes, "loopId")(pause(gracePeriod).exec(exec(imap("deleteFolder").deleteFolder(s"$mailboxDNewName.$${loopId}").check(ok)))))
```

Comment:
So it seems in IMAP if we delete a folder, it retroactively deletes all subfolders... (or so I think) so why do you need to try delete those subfolders after? those that work?

Reply:
> those that work?

Deleting subfolders explicitly did work on James.

Reply:
Ok... not sure to remember how it works then :D

Reply:
When you don't know read the spec, Luke!

```
      The DELETE command MUST NOT remove inferior hierarchical names.
      For example, if a mailbox "foo" has an inferior "foo.bar"
      (assuming "." is the hierarchy delimiter character), removing
      "foo" MUST NOT remove "foo.bar".  It is an error to attempt to
      delete a name that has inferior hierarchical names and also has
      the \Noselect mailbox name attribute (see the description of the
      LIST response for more details).
  ```